src/properties.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * Property field types
+ * examples are propvided for the different types:
+ *
+ */
+export const FieldTypes = {
+  /**
+   * userName: {
+   *   type: csf.FieldTypes.TEXT,
+   *   label: 'Name',
```

Comment:
I'd prefer the term 'title' OR 'name'

Reply:
name is already used to refer to the property name, but title is available and I can rename it?

Reply:
Just to keep in mind, in knobs-1 its `label`

Reply:
yes, I know, we are not required to make this match that api.

The word "label" stems from the fact we're showing a `<label />` but that's an implementation detail. I'd rather make an attempt to remove confusing language, and name things the same if at all possible

Reply:
Maybe its me, but title is confusing for me :) Lets get more feedback, but on my end I would rather not change this.

Reply:
I don't much like "label". I'd much rather this be "name" (which represents a human readable string of any characters) and have some other property for what is currently name. Maybe it's:

["name", "label"] -> ["id", "name"] 

?

Reply:
name is the name of the variable that is passed to the story 

for example 
```
export const textStory = ({ text }) => text;
  
textStory.story = {
  properties: {
     text: { type: PropertyTypes.TEXT, label: 'Text' }
  }
}
```

here name is the `text` variable name. If we rename the label to name:
```
     text: { type: PropertyTypes.TEXT, name: 'Text}
```

now would the user expect to receive { Text } as the prop to the story
Anyway for me - we say variable 'name',  not variable 'id' and I will keep confusing what to refer to for this.

Reply:
I can live with "name" and "label" I guess. The asymmetry with `CSFStory` feels really confusing to me. 

wouldn't it to be better like this:

export const textStory = ({ text }) => text;
  
textStory.story = {
  label: 'Text Story`, 
  properties: {
     text: { type: PropertyTypes.TEXT, label: 'Text' }
  }
}
```

then we'd say "The story with name 'textStory'"

(on a similar note that we don't have a better name that 'kind' for collections of stories feels super weird)

Reply:
@shilman - last call - name or label?

Reply:
I think it was an intentional choice that the "kind" (or "component") has a *title* but the story has a *name* -- it makes it simpler to talk about these things and not have to qualify them all the time.

With that in mind it's kind of nice that there's another term for the name of a property.